backend/FileSystem/FileManager.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\FileSystem;
+
+use Exception;
+
+class FileManager
+{
+    public function __construct(
+        protected DirectoryManager &$directoryManager
```

Comment:
Why do you expecting explicite reference to the directory manager here?

Reply:
I was worried the object will be copied over, as I have not familiarized with PHP that much yet. Might be a bad habit from some issue I had in the past with something else.